c10/core/ScalarType.h

```diff
@@ -40,7 +40,8 @@ _(at::Half,Half,d) \
 _(float,Float,d)   \
 _(double,Double,d) \
 _(std::complex<float>,ComplexFloat,z) \
-_(std::complex<double>,ComplexDouble,z)
+_(std::complex<double>,ComplexDouble,z) \
+_(bool,Bool,i)
 
 #define AT_FORALL_SCALAR_TYPES(_) \
```

Comment:
I'm not very confident about the lack of Bool here -- can you look at the call sites and see what makes sense to do here?